packages/terra-dropdown-button/src/SplitButton.jsx

```diff
@@ -0,0 +1,181 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames/bind';
+import KeyCode from 'keycode-js';
+import { injectIntl, intlShape } from 'react-intl';
+import DropdownButtonBase from './_DropdownButtonBase';
+import styles from './SplitButton.module.scss';
+import Button from './Button';
+
+const cx = classNames.bind(styles);
+
+const Variants = {
+  NEUTRAL: 'neutral',
+  GHOST: 'ghost',
+};
+
+const propTypes = {
+  /**
+   * The options to display in the dropdown. Should be comprised of the subcomponent `Button`.
+   */
+  children: PropTypes.node.isRequired,
+  /**
+   * Determines whether the primary button and expanding the dropdown should be disabled.
+   */
+  disabled: PropTypes.bool,
+  /**
+   * Determines whether the component should have block styles applied. The dropdown will match the component's width.
+   */
+  isBlock: PropTypes.bool,
+  /**
+   * Sets the text that will be shown on the primary button which is outside the dropdown.
+   */
+  primaryOptionLabel: PropTypes.string.isRequired,
+  /**
+   * What will be called when the primary button is pressed.
+   */
+  onClick: PropTypes.func.isRequired,
+  /**
+   * Sets the styles of the component.  The default is 'neutral'. Must be either 'neutral' or 'ghost'.
+   */
+  variant: PropTypes.oneOf(Object.values(Variants)),
+  /**
+   * @private
+   * The intl object to be injected for translations.
+   */
+  intl: intlShape.isRequired,
+};
+
+const defaultProps = {
+  disabled: false,
+  isBlock: false,
+  variant: 'neutral',
+};
+
+class SplitButton extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.handleDropdownButtonClick = this.handleDropdownButtonClick.bind(this);
+    this.handleDropdownRequestClose = this.handleDropdownRequestClose.bind(this);
+    this.handlePrimaryKeyDown = this.handlePrimaryKeyDown.bind(this);
+    this.handlePrimaryKeyUp = this.handlePrimaryKeyUp.bind(this);
+    this.handleCaretKeyDown = this.handleCaretKeyDown.bind(this);
+    this.handleCaretKeyUp = this.handleCaretKeyUp.bind(this);
+
+    this.state = { isOpen: false, caretActive: false, primaryActive: false };
```

Comment:
nit, for clarity, can we add 'is' to these state attributes
```suggestion
    this.state = { isOpen: false, caretIsActive: false, primaryIsActive: false };
```